lib/spack/spack/cmd/buildcache.py

```diff
@@ -210,8 +210,7 @@ def createtarball(args):
             specs.add(match)
             tty.msg('recursing dependencies')
             for d, node in match.traverse(order='post',
-                                          depth=True,
-                                          deptype=('link', 'run')):
```

Comment:
(Question) Is there a reason you chose to have `createtarball` package up all the build dependencies too, and not adjust `installtarball` to skip build-only dependencies?

Reply:
I should take the time to figure it out.